decoder/expression_candidates.go

```diff
@@ -331,6 +331,10 @@ func (d *PathDecoder) constraintToCandidates(ctx context.Context, constraint sch
 		if schema.ActiveCountFromContext(ctx) && attr.Name != "count" {
 			candidates = append(candidates, countIndexCandidate(editRng))
 		}
+		if schema.ActiveForEachFromContext(ctx) {
```

Comment:
Since it also doesn't make sense to complete `each.*` within `for_each = HERE`, similar to how we don't complete `count.index` in `count = HERE` I think we'll need some check for that + ideally a test too.